Fix: Add flag to detect when the input field is active

danilminibaev opened the pull request:
This fixes problem that the hex input field value was not updating when `onChange(newColor)` was triggered from the outside.

The reason for that was that the value passed to the input field was derived from the `intermediateHexInput` value directly like here `const hexColorString = intermediateHexInput === null ? this.color.hex() : intermediateHexInput`

That was made for the fallback to the last correct hex value in case the user changing the color manually from the field, but when `onChange` was triggered with some color value since the `intermediateHexInput` was not null it was assigning value from it all the time.

Now we just checking if the input value changing (so we can go to fallback if yes, but if not - then just set current actual value)

Commit:
Fix: Add flag to detect when the input field is active

UNITY-4972

danilminibaev committed:
Fix: Space line added

Commit:
Fix: Change commentary, attempt to remove trailing space warning

ingmarh commented on src/form/ColorpickerField.jsx:
```diff
@@ -121,6 +121,8 @@ export class ColorpickerFieldRaw extends Component {
     intermediateHexInput: null
   }
 
+  changedByInput = false
```
I'd remove this line because the only purpose of this instance variable is to be able to test if its value is truthy or falsy inside `render()`. If the value is `undefined`, it's still falsy (and it's kind of a "temporary" variable anyway).

I think it's enough to set the variable's value to `true`/`false` inside the hex input's `onChange` handler.

danilminibaev replied:
As discussed, will remove it 👍

ingmarh commented on src/form/ColorpickerField.jsx:
```diff
@@ -223,7 +225,12 @@ export class ColorpickerFieldRaw extends Component {
             const {value} = e.target
 
-            this.setState({intermediateHexInput: value})
+            this.changedByInput = true
```
I suggest to change the naming to something more descriptive, e.g. `this.hexInputValueChange`.

danilminibaev replied:
Good!

ingmarh commented on src/form/ColorpickerField.jsx:
```diff
@@ -223,7 +225,12 @@ export class ColorpickerFieldRaw extends Component {
+            this.changedByInput = true
+
+            this.setState({intermediateHexInput: value}, () => {
+              // after color was changed we can set it to initial value
```
I think this comment is not necessary, as it is obvious that the variable is reset from `true` to `false`. (If I didn't misunderstood this comment.)

danilminibaev replied:
Oh, nice then

ingmarh commented on src/form/ColorpickerField.jsx:
```diff
@@ -169,8 +171,8 @@ export class ColorpickerFieldRaw extends Component {
 
-    const hexColorString = intermediateHexInput === null
-      ? this.color.hex()
+    const hexColorString = (intermediateHexInput === null || !this.changedByInput)
```
I think this can be simplified to:

```jsx
const hexColorString = this.changedByInput ? intermediateHexInput : this.color.hex()
```

danilminibaev replied:
Oh, good, will change that

ingmarh commented on package-lock.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@epages/react-components",
-  "version": "0.6.1",
+  "version": "0.6.2",
```
Please update or rebase this branch onto "master".
It looks like this PR is based on an old state of the "master" branch (the current version in "master" is `0.9.0`).

ingmarh reviewed (changes requested):
Please also update the commit message and/or the description of this PR to describe which problem this change solves.

danilminibaev committed:
Fix: Address review annotations

 - Meaningful name for changing hex input flag variable
 - Removed obsolete commentary

Commit:
Merge remote-tracking branch 'upstream/master' into fix/color-picker-reset
